dll/shellext/zipfldr/zipfldr.cpp

```diff
@@ -174,16 +174,17 @@ STDAPI DllUnregisterServer()
 
 EXTERN_C
 BOOL WINAPI
-RouteTheCall(
-    IN HWND hWndOwner,
-    IN HINSTANCE hInstance,
-    IN PCWSTR lpStringArg,
-    IN INT Show)
+RouteTheCallW(IN HWND hWndOwner, IN HINSTANCE hInstance, IN PCWSTR lpStringArg, IN INT Show)
 {
     CStringW path = lpStringArg;
     PathRemoveBlanksW(path.GetBuffer());
     path.ReleaseBuffer();
     path = L"\"" + path + L"\"";
-    ShellExecuteW(NULL, L"open", L"explorer.exe", path.GetString(), NULL, SW_SHOWNORMAL);
+
+    WCHAR app[MAX_PATH];
+    GetWindowsDirectoryW(app, _countof(app));
+    PathAppendW(app, L"explorer.exe");
```

Comment:
What would happen if you wouldn't pass the full path to explorer? If you wouldn't really be able to point out a specific benefit, then I would vote for the shortest solution that would do the job.

Reply:
Not passing the full path and just hoping is bad if

- There is a Explorer.exe in the current directory
- Something is wrong with %path%

Reply:
In other words: Just the pure fix for "zipfldr not opening a zip via ShellExecute" would just be the thing in the spec file. The rest here can be understood as 'other security related improvements'. Correct?

Reply:
This rundll32 mechanism is already fragile, no need to make it worse with weird breakage if it can't find Explorer.